src/cmap/test/mod.rs

```diff
@@ -435,3 +438,44 @@ async fn cmap_spec_tests() {
 
     run_spec_test(&["connection-monitoring-and-pooling"], run_cmap_spec_tests).await;
 }
+
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+async fn redact_credential() {
+    let _lock = LOCK.run_concurrently().await;
+
+    let client = TestClient::new().await;
+    if !client.auth_enabled() {
+        println!("skipping redact_credential due to auth not being enabled");
+        return;
+    }
+
+    let mut options = CLIENT_OPTIONS.clone();
+    options.direct_connection = Some(true);
+    options.hosts.drain(1..);
+
+    let credential = options.credential.clone().unwrap();
+
+    let handler = Arc::new(crate::test::EventHandler::new());
+    let mut subscriber = handler.subscribe();
+    options.cmap_event_handler = Some(handler.clone());
+
+    let _client = Client::with_options(options).unwrap();
+
+    subscriber
+        .wait_for_event(EVENT_TIMEOUT, |e| {
+            if let crate::test::Event::CmapEvent(CmapEvent::ConnectionPoolCreated(event)) = e {
```

Comment:
is it worth also explicitly checking that REDACTED is in the debug output?

Reply:
Updated to check.